src/HtmlReader/index.tsx

```diff
@@ -1,396 +1,262 @@
-import D2Reader from '@d-i-t-a/reader';
 import React from 'react';
 import {
   ColorMode,
-  HtmlReaderState,
   ReaderReturn,
   ReaderArguments,
   FontFamily,
+  HtmlNavigator,
 } from '../types';
-import HtmlReaderContent from './HtmlReaderContent';
-import { Locator } from '@d-i-t-a/reader';
-import '../../node_modules/@d-i-t-a/reader/dist/reader.css';
+import LoadingSkeleton from '../ui/LoadingSkeleton';
+import { DEFAULT_HEIGHT, DEFAULT_SHOULD_GROW_WHEN_SCROLLING } from '..';
 import {
-  DEFAULT_HEIGHT,
-  DEFAULT_SHOULD_GROW_WHEN_SCROLLING,
-  HEADER_HEIGHT,
-} from '../constants';
-import {
-  GetContent,
-  Injectable,
-  NavigatorAPI,
-} from '@d-i-t-a/reader/dist/types/navigator/IFrameNavigator';
-import debounce from 'debounce';
-
-type HtmlState = HtmlReaderState & {
-  reader: D2Reader | undefined;
-  location: undefined | Locator;
-};
+  fetchAsTxt,
+  calcPosition,
+  defaultInjectables,
+  defaultInjectablesFixed,
+  isSameResource,
+} from './lib';
+import makeHtmlReducer, { inactiveState } from './reducer';
+import { navigateToHash, navigateToProgression } from './effects';
+import useResource from './useResource';
+import useLocationQuery from './useLocationQuery';
+import useWindowResize from './useWindowResize';
+import { useUpdateScroll } from './useUpdateScroll';
+import useUpdateCSS from './useUpdateCSS';
+import useIframeLinkClick from './useIframeLinkClick';
 
 /**
- * If we provide injectables that are not found, the app won't load at all.
- * Therefore we will not provide any default injectables.
+ * @TODO :
+ *
+ * - WORKING ON
+ *  - provide default injectables (Readium CSS)
+ *  - fix navigate to hash.
+ *
+ * Future:
+ *  - Don't use ReadiumCSS for fixed layout
```

Comment:
Why?

Reply:
It doesn't support fixed layout. Instead you rely on the publisher's CSS